tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -366,6 +366,17 @@ Status ExecuteThunks(const std::string& module_name,
         *run_options, buffer_allocations, main_stream,
         async_comms_stream.ok() ? async_comms_stream->get() : nullptr};
     TF_RETURN_IF_ERROR(thunk->ExecuteOnStream(thunk_params));
+
+    // GPU multi-stream is currently not supported. We need to make
+    // exeuctions on async_comms_stream blocking to avoid race condition.
+    if (async_comms_stream.ok() && NeedsAsyncCommsStream(*thunk)) {
+      Status block_status = async_comms_stream->get()->BlockHostUntilDone();
```

Comment:
Just `TF_RETURN_IF_ERROR(async_comms_stream->get()->BlockHostUntilDone())`? We can pipe extra info to that, e.g. `TF_RETURN_IF_ERROR(...) << "extra logging"`

Reply:
Changed.